### ISSUE TITLE
This cookbook is not fully compatible with CentOS 7

### ISSUE THREAD

**Issue by tgagor** [rackspace-cookbooks/varnish#62]
Varnish is installed but configuration is generated in bad place.

I was installing Varnish 4 on CentOS 7.0 and found that configuration changes in /etc/sysconfig/varnish do not apply to varnish service. Ex. I switched storage to malloc but processes were still run with file:

```
# grep '\-s' /etc/sysconfig/varnish 
              -s malloc,3G \
```
but:
```
# ps axu | grep varnish
root     26586  0.0  2.1 124776 84340 ?        SLs  10:41   0:00 /usr/sbin/varnishd -P /var/run/varnish.pid -f /etc/varnish/default.vcl -a :6081 -T 127.0.0.1:6082 -t 120 -u varnish -g varnish -S /etc/varnish/secret -s file,/var/lib/varnish/varnish_storage.bin,1G
```

On CentOS 7 systemd configuration for services are kept in:
```
# rpm -ql varnish | grep systemd
/usr/lib/systemd/system/varnish.service
/usr/lib/systemd/system/varnishlog.service
/usr/lib/systemd/system/varnishncsa.service
```
And varnish.service here looks like:
```
[Unit]
Description=Varnish a high-perfomance HTTP accelerator
After=syslog.target network.target

[Service]

#
# If you want to make changes to this file, please copy it to 
# /etc/systemd/system/varnish.service and make your changes there.
# This will override the file kept at /lib/systemd/system/varnish.service
#
# Enviroment variables may be found in /etc/varnish/varnish.params
#

# Maximum number of open files (for ulimit -n)
LimitNOFILE=131072

# Locked shared memory (for ulimit -l)
# Default log size is 82MB + header
LimitMEMLOCK=82000

# Maximum size of the corefile. 
LimitCORE=infinity

EnvironmentFile=/etc/varnish/varnish.params

Type=forking
PIDFile=/var/run/varnish.pid
PrivateTmp=true
ExecStartPre=/usr/sbin/varnishd -C -f $VARNISH_VCL_CONF
ExecStart=/usr/sbin/varnishd \
        -P /var/run/varnish.pid \
        -f $VARNISH_VCL_CONF \
        -a ${VARNISH_LISTEN_ADDRESS}:${VARNISH_LISTEN_PORT} \
        -T ${VARNISH_ADMIN_LISTEN_ADDRESS}:${VARNISH_ADMIN_LISTEN_PORT} \
        -t $VARNISH_TTL \
        -u $VARNISH_USER \
        -g $VARNISH_GROUP \
        -S $VARNISH_SECRET_FILE \
        -s $VARNISH_STORAGE \
        $DAEMON_OPTS

ExecReload=/usr/sbin/varnish_reload_vcl

[Install]
WantedBy=multi-user.target
```

So this file is preconfigured from: `/etc/varnish/varnish.params`

```
# Varnish environment configuration description. This was derived from
# the old style sysconfig/defaults settings

# Set this to 1 to make systemd reload try to switch vcl without restart.
RELOAD_VCL=1

# Main configuration file. You probably want to change it.
VARNISH_VCL_CONF=/etc/varnish/default.vcl

# Default address and port to bind to. Blank address means all IPv4
# and IPv6 interfaces, otherwise specify a host name, an IPv4 dotted
# quad, or an IPv6 address in brackets.
# VARNISH_LISTEN_ADDRESS=192.168.1.5
VARNISH_LISTEN_PORT=6081

# Admin interface listen address and port
VARNISH_ADMIN_LISTEN_ADDRESS=127.0.0.1
VARNISH_ADMIN_LISTEN_PORT=6082

# Shared secret file for admin interface
VARNISH_SECRET_FILE=/etc/varnish/secret

# Backend storage specification, see Storage Types in the varnishd(5)
# man page for details.
VARNISH_STORAGE="file,/var/lib/varnish/varnish_storage.bin,1G"

# Default TTL used when the backend does not specify one
VARNISH_TTL=120

# User and group for the varnishd worker processes
VARNISH_USER=varnish
VARNISH_GROUP=varnish

# Other options, see the man page varnishd(1)
#DAEMON_OPTS="-p thread_pool_min=5 -p thread_pool_max=500 -p thread_pool_timeout=300"
```

So for now support for CentOS 7 in your cookbook looks to be limited to Varnish installation :(
Could you update this configuration changes? Will it help if I prepare pull request with some changes?

**Comment by jujugrrr** [comment]
> Will it help if I prepare pull request with some changes?

hi @gagor . It will indeed! If you have a PR ready it will be faster (especially if you have tests)

**Comment by gagor** [comment]
I don't have anything for now but I have to face this problem on two varnish instances so I may prepare something that you may use. I hope to get this working later this day.

**Comment by jujugrrr** [comment]
thanks a lot @gagor

**Comment by gagor** [comment]
I prepared something that at least will do most of the work for now.

**Comment by gsreynolds** [comment]
I believe this should be addressed in #80 once it's merged

**Comment by martinb3** [comment]
#80 has been merged. Going to close this.
